addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -43,6 +43,509 @@ var globalSelector = {
+
+            // Placing it in the top left corner
+            this.options.wysiwyg.odooEditor.observerActive(this.observerName);
+            this.draggedItemEl.style.gridArea = `1 / 1 / ${1 + this.dragState.columnRowCount} / ${1 + this.dragState.columnColCount}`;
+            const rowCount = Math.max(rowEl.dataset.rowCount, this.dragState.columnRowCount);
+            rowEl.dataset.rowCount = rowCount;
+            this.options.wysiwyg.odooEditor.observerUnactive(this.observerName);
+
+            // Setting the grid item z-index
+            if (rowEl === this.dragState.startingGrid) {
+                this.draggedItemEl.style.zIndex = this.dragState.startingZIndex;
+            } else {
+                gridUtils._setElementToMaxZindex(this.draggedItemEl, rowEl);
+            }
+        } else if (this.draggedItemEl.classList.contains("o_grid_item")) {
+            // Case when a grid column is dropped near a non-grid dropzone
+            this.options.wysiwyg.odooEditor.observerActive(this.observerName);
+            gridUtils._convertToNormalColumn(this.draggedItemEl);
+            this.options.wysiwyg.odooEditor.observerUnactive(this.observerName);
+        }
+        this.dragState.currentDropzoneEl = dropzoneEl;
+    }
+    /**
+     * Handles the insertion of an element in a dropzone.
+     *
+     * @param {Object} dropzone - The dropzone over which the element is dragged
+     * with its position and dimension information.
+     */
+    dropzoneOver(dropzone) {
+        const dropzoneEl = dropzone.el;
+        if (this.isDropped()) {
+            // Checking if the "out" event happened before this "over":
+            // if `isDropped()` is "true", "out" didn't happen because it sets
+            // it to "false". We are therefore in the case of an "over" after an
+            // "over" and we need to escape the previous dropzone first.
+            this.draggedItemEl.remove();
+            if (this.itemComesFromRightPanel) {
+                this._getOriginalDraggedItemEl().classList.add("oe_snippet_body");
+            }
+            this._outPreviousDropzone(dropzoneEl);
+        }
+
+        // Prevent a column to be trapped in an upper grid dropzone at the start
+        // of the drag.
+        if (this.dragState.isColumn && this.dragState.overFirstDropzone) {
+            this.dragState.overFirstDropzone = false;
+
+            // The column is considered as glued to the dropzone if the dropzone
+            // is above and if the space between them is less than 25px (the
+            // move handle height is 22px so 25 is a safety margin).
+            const columnTop = this.dragState.columnTop;
+            const dropzoneBottom = dropzoneEl.getBoundingClientRect().bottom;
+            const areDropzonesGlued = (columnTop >= dropzoneBottom) && (columnTop - dropzoneBottom < 25);
+
+            if (areDropzonesGlued && dropzoneEl.classList.contains("oe_grid_zone")) {
+                return;
+            }
+        }
```

Comment:
If am not mistaken, I think that the "over > over > out" case will never happen with the glued dropzone case : the second one is only checked at the first over of a dropzone; at this moment, an over after an over is impossible :) 
The only reason of why I did it is to be clean : since the second commit, `this.dropped` is equivalent to `!!this.dragState.currentDropzoneEl`. If I did not gather those two parts together, the code would have look like:
```js
 if (this.isDropped()) {
   // Checking if the "out" event happened before this "over":
   // if `isDropped()` is "true", "out" didn't happen because it sets
   // it to "false". We are therefore in the case of an "over" after an
   // "over" and we need to escape the previous dropzone first.
   this.draggedItemEl.remove();
}
// Prevent a column to be trapped in an upper grid dropzone at the start
// of the drag.
if (this.dragState.isColumn && this.dragState.overFirstDropzone) {
   ...
}
 if (this.isDropped()) {
   this._outPreviousDropzone(dropzoneEl);
}
```